ZeeHomeScreen/Classes/Components/Carousel/CarouselViewController.m

```diff
@@ -282,20 +282,19 @@ - (NSInteger)numberOfItemsInPromotionView:(APPromotionView *)promotionView {
 - (UICollectionViewCell *)promotionView:(APPromotionView *)promotionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
     
     CellModel *cellModel = self.dataSource[indexPath.row];
-    
-    NSString *reuseIdentifier = [cellModel.identifier stringByAppendingFormat:@"_%@", cellModel.layoutStyle];
-    
+
     UniversalCollectionViewCell *cell = [promotionView dequeueReusableCellWithReuseIdentifier:cellModel.layoutStyle
                                                                                  forIndexPath:indexPath];
     CGFloat width = [UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPhone ? cellModel.iphoneWidth : cellModel.ipadWidth;
     CGFloat height = [UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPhone ? cellModel.iphoneHeight : cellModel.ipadHeight;
     cell.frame = CGRectMake(cell.frame.origin.x, cell.frame.origin.y, width, height);
     if (cell.componentViewController == nil) {
-        cell.componentViewController = [ComponenttFactory componentViewControllerWithComponentModel:cellModel
-                                                                                           andModel:cellModel.entry
-                                                                                            forView:cell.contentView
-                                                                                           delegate:self.delegate
-                                                                               parentViewController:self];
+        UIViewController<ComponentProtocol> *componentViewController = [ComponenttFactory viewControllerForComponentModel:cellModel];
+        if (componentViewController != nil) {
+            [self addChildViewController:componentViewController];
```

Comment:
This was handled previously by factory method, I had to change it to comply to what Apple documentation says. Factory method forced the order of execution as:
-addSubview
-addChildViewController
-didMoveToParentViewController

But the correct order according to Apple should be:
-addChildViewController
-addSubview
-didMoveToParentViewController

Because `addChildViewController` already handles the case with calling `will` method, which might do things on a previously attached view, which in the other case was already altered.

Reply:
SRP fix.